### PR TITLE
Add Azure activity logs to e2e artifacts

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**
/kind other

**What this PR does / why we need it**: Adds Azure activity logs to E2E artifacts to facilitate debugging. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**: This is still work in progress, mostly opening now for testing. Still need to add more event data / format and organize the code.

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Add activity logs to e2e artifacts
```

### CODE REVIEW

File: test/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -149,6 +154,50 @@ func (acp *AzureClusterProxy) CollectWorkloadClusterLogs(ctx context.Context, na
 			}(pod, container)
 		}
 	}
+
+	Byf("Dumping workload cluster %s/%s Azure activity log", namespace, name)
```
Comment (CecileRobertMichon) [inline test/e2e/e2e_suite_test.go:158]: I'll probably refactor this into its own func

Comment (devigned) [reply test/e2e/e2e_suite_test.go:158]: The kube-system logger and the events should both be funcs. I should have extracted the func in my previous PR.

File: test/e2e/azure_privatecluster.go [hunk 2]
```diff
@@ -21,8 +21,8 @@ package e2e
 import (
 	"context"
 	"fmt"
-	"github.com/Azure/azure-sdk-for-go/profiles/latest/network/mgmt/network"
-	"github.com/Azure/azure-sdk-for-go/profiles/latest/resources/mgmt/resources"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-06-01/network"
```
Comment (devigned) [inline test/e2e/azure_privatecluster.go:24]: :+1:

**Review by devigned — CHANGES_REQUESTED** [review]

**Comment by CecileRobertMichon** [comment]
This is what activity logs look like https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/1136/pull-cluster-api-provider-azure-e2e/1352943135788896256/artifacts/clusters/capz-e2e-08t0ut/azure-activity-logs/capz-e2e-08t0ut.log

**Comment by CecileRobertMichon** [comment]
`pull-cluster-api-provider-azure-e2e` failed because of a time out during pod log collection I believe... I wonder if we should make it best effort and timeout so it doesn't fail the overall test if log collection fails?

**Review by devigned — COMMENTED** [review]
What do you think about outputting as json or yaml?

**Comment by CecileRobertMichon** [comment]
> failed because of a time out during pod log collection I believe

I take that back, it's actually the overall test that timed out after 2 hours (I guess because of the large number of clusters we are deploying)

**Review by devigned — APPROVED** [review]
On second thought, let's get this in and follow up with formatting if we need it.

/lgtm
/assign @nader-ziada

**Comment by nader-ziada** [comment]
this applies to the capi tests as well, right?

**Comment by devigned** [comment]
> this applies to the capi tests as well, right?

Yes, it should, @nader-ziada. We proxy the `ClusterProxy` and intercept the calls to`CollectWorkloadClusterLogs`.

**Comment by CecileRobertMichon** [comment]
/hold

it looks like tests are taking way longer now, will investigate

**Comment by CecileRobertMichon** [comment]
Added duration logs, looks like it's only taking ~1s to get activity logs `[1] STEP: Fetching activity logs took 1.338427791s` (and a bit less for pod logs `[1] STEP: Fetching kube-system pod logs took 701.418768ms`)

Will monitor any timeouts but doesn't look like the logs is what is causing an increased duration

/hold cancel

**Comment by CecileRobertMichon** [comment]
/hold

something is going on, test is still timing out

**Comment by CecileRobertMichon** [comment]
this is what it looks like now https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/1136/pull-cluster-api-provider-azure-e2e-windows/1354931982861078528/artifacts/clusters/capz-e2e-1vuya5/azure-activity-logs/capz-e2e-1vuya5.log

**Comment by CecileRobertMichon** [comment]
different provisioning flake 😕 

/retest

**Pull request merged**